libpromises/evalfunction.c

```diff
@@ -789,6 +785,122 @@ static FnCallResult FnCallIP2Host(ARG_UNUSED EvalContext *ctx,
 
 /*********************************************************************/
 
+static FnCallResult FnCallSysctlValue(ARG_UNUSED EvalContext *ctx,
+                                      ARG_UNUSED ARG_UNUSED const Policy *policy,
+                                      const FnCall *fp,
+                                      const Rlist *finalargs)
+{
+#ifdef __linux__
+    const bool sysctlvalue_mode = (strcmp(fp->name, "sysctlvalue") == 0);
+
+    size_t max_sysctl_data = 16 * 1024;
+    Buffer *procrootbuf = BufferNew();
+    // Assumes that FILE_SEPARATOR is /
+    BufferAppendString(procrootbuf, GetRelocatedProcdirRoot());
+    BufferAppendString(procrootbuf, "/proc/sys");
+
+    if (sysctlvalue_mode)
+    {
+        Buffer *key = BufferNewFrom(RlistScalarValue(finalargs),
+                                    strlen(RlistScalarValue(finalargs)));
+
+        Buffer *filenamebuf = BufferCopy(procrootbuf);
```

Comment:
You are not using `procrootbuf` again inside this if statement, and all code paths lead to a return. I suggest making `filenamebuf` an alias to it, instead of a copy, or to simply use `procrootbuf` throughout. This avoids the whole copy.